osbenchmark/aggregator.py

```diff
@@ -194,18 +198,28 @@ def calculate_weighted_average(self, task_metrics: Dict[str, List[Any]], iterati
                         if iterations > 1:
                             weighted_sum = sum(value * iterations for value in item_values)
                             total_iterations = iterations * len(values)
-                            weighted_metrics[metric][item_key] = weighted_sum / total_iterations
+                            weighted_avg = weighted_sum / total_iterations
                         else:
-                            weighted_metrics[metric][item_key] = sum(item_values) / len(item_values)
+                            weighted_avg = sum(item_values) / len(item_values)
+                        weighted_metrics[metric][item_key] = weighted_avg
             else:
                 if iterations > 1:
                     weighted_sum = sum(value * iterations for value in values)
                     total_iterations = iterations * len(values)
-                    weighted_metrics[metric] = weighted_sum / total_iterations
+                    weighted_avg = weighted_sum / total_iterations
                 else:
-                    weighted_metrics[metric] = sum(values) / len(values)
+                    weighted_avg = sum(values) / len(values)
+                weighted_metrics[metric] = weighted_avg
```

Comment:
Why the special case for `iterations == 1`?  Doesn't this reduce to the same computation?  And if the iteration number is the same for every element in values, the special-casing might not be needed.

Reply:
This is fixed, but I want to include support for custom iteration values in the future so I think it should stay for now.